ts/output/common/FontData.ts

```diff
@@ -673,27 +804,63 @@ export class FontData<C extends CharOptions, V extends VariantData<C>, D extends
    * @constructor
    */
   constructor(options: OptionList = null) {
-    let CLASS = (this.constructor as typeof FontData);
+    let CLASS = this.CLASS;
     this.options = userOptions(defaultOptions({}, CLASS.OPTIONS), options);
     this.params = {...CLASS.defaultParams};
     this.sizeVariants = [...CLASS.defaultSizeVariants];
     this.stretchVariants = [...CLASS.defaultStretchVariants];
-    this.cssFontMap = {...CLASS.defaultCssFonts};
-    for (const name of Object.keys(this.cssFontMap)) {
-      if (this.cssFontMap[name][0] === 'unknown') {
-        this.cssFontMap[name][0] = this.options.unknownFamily;
-      }
-    }
+    this.defineCssFonts(CLASS.defaultCssFonts);
     this.cssFamilyPrefix = CLASS.defaultCssFamilyPrefix;
     this.createVariants(CLASS.defaultVariants);
-    this.defineDelimiters(CLASS.defaultDelimiters);
-    for (const name of Object.keys(CLASS.defaultChars)) {
-      this.defineChars(name, CLASS.defaultChars[name]);
-    }
+    this.defineDelimiters(CLASS.defaultDelimiters as DelimiterMap<D>);
+    Object.keys(CLASS.defaultChars).forEach(name => this.defineChars(name, CLASS.defaultChars[name] as CharMap<C>));
     this.defineRemap('accent', CLASS.defaultAccentMap);
     this.defineRemap('mo', CLASS.defaultMoMap);
     this.defineRemap('mn', CLASS.defaultMnMap);
     this.defineDynamicCharacters(CLASS.dynamicFiles);
+    CLASS.dynamicExtensions.forEach(data => this.defineDynamicCharacters(data.files));
+  }
+
+  /**
+   * Add an extension to an existing font instance (options will get their defaults).
+   *
+   * @param {FontExtensionData} data    The data for the font extension to merge into this font.
+   * @param {string} prefix             The [prefix] to add to all component names
+   */
+  public addExtension(data: FontExtensionData<C, D>, prefix: string = '') {
+    const dynamicFont = {
+      name: data.name,
+      prefix: prefix,
+      files: this.CLASS.defineDynamicFiles(data.ranges, prefix),
+      sizeN: this.sizeVariants.length,
+      stretchN: this.stretchVariants.length
+    };
+    this.CLASS.dynamicExtensions.set(data.name, dynamicFont);
+
+    data.options && defaultOptions(this.options, data.options);
+    data.parameters && defaultOptions(this.params, data.parameters);
+    if (data.sizeVariants) {
+      this.sizeVariants = mergeOptions(this.sizeVariants, data.sizeVariants);
```

Comment:
Since `mergeOptions` is used only in this module, why not handle the empty case there as it would make the code a lot cleaner. 

It should not have repercussions for other methods, such as `defineCssFonts`. `createVariants` might need a guard.  And `this.sizeVariants` is always a list anyway.

Reply:
Changes made.